app/src/main/java/de/langerhans/odintools/main/MainViewModel.kt

```diff
@@ -170,6 +170,47 @@ class MainViewModel @Inject constructor(
         }
     }
 
+    fun remapButtonClicked(setting: String) {
+        _uiState.update {
+            it.copy(
+                showRemapButtonDialog = true,
+                currentButtonSetting = setting,
+                currentButtonKeyCode = executor.getSystemSetting(setting, 0)
+            )
+        }
+    }
+
+    fun remapButtonDialogDismissed() {
+        _uiState.update {
+            it.copy(showRemapButtonDialog = false)
+        }
+    }
+
+    private fun getDefaultKeyCode(setting: String): Int {
+        if (setting == "remap_custom_to_m1_value")
```

Comment:
Please add braces even to single if statements, I just think it looks cleaner :D